src/app/components/chat/room-storage/room-storage-service.js

```diff
@@ -129,7 +153,15 @@ angular.module( 'App.Chat' ).service( 'Chat_RoomStorage', function( $injector, $
 
 		var roomIndex = data.rooms.indexOf( roomId );
 		if ( roomIndex !== -1 ) {
-			data.rooms.splice( roomIndex, 1 );
```

Comment:
@hworld If they for some reason have roomId in their `rooms` array more than once, this will clear out all of them. Just testing for all the edge cases.